flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamPhysicalMultiJoin.java

```diff
@@ -0,0 +1,221 @@
+            final List<RelHint> hints) {
+        super(cluster, traitSet);
+        this.inputs = inputs;
+        this.rowType = rowType;
+        this.joinFilter = joinFilter;
+        this.joinTypes = joinTypes;
+        this.joinConditions = joinConditions;
+        this.joinAttributeMap = joinAttributeMap;
+        this.postJoinFilter = postJoinFilter;
+        this.hints = hints;
+    }
+
+    @Override
+    public boolean requireWatermark() {
+        return false;
+    }
+
+    @Override
+    public List<RelNode> getInputs() {
+        return inputs;
+    }
+
+    @Override
+    public void replaceInput(final int ordinalInParent, final RelNode p) {
+        final List<RelNode> newInputs = new ArrayList<>(inputs);
+        newInputs.set(ordinalInParent, p);
+        this.inputs = List.copyOf(newInputs);
+        recomputeDigest();
+    }
+
+    @Override
+    public RelNode copy(final RelTraitSet traitSet, final List<RelNode> inputs) {
+        return new StreamPhysicalMultiJoin(
+                getCluster(),
+                traitSet,
+                inputs,
+                joinFilter,
+                getRowType(),
+                joinConditions,
+                joinTypes,
+                joinAttributeMap,
+                postJoinFilter,
+                hints);
+    }
+
+    @Override
+    public RelOptCost computeSelfCost(final RelOptPlanner planner, final RelMetadataQuery mq) {
+        final double elementRate = 100.0d * getInputs().size();
+        return planner.getCostFactory().makeCost(elementRate, elementRate, 0);
+    }
+
+    @Override
+    public RelWriter explainTerms(final RelWriter pw) {
+        super.explainTerms(pw);
+        for (final Ord<RelNode> ord : Ord.zip(inputs)) {
+            pw.input("input#" + ord.i, ord.e);
+        }
+        return pw.item("joinFilter", joinFilter)
+                .item("joinTypes", joinTypes)
+                .item("joinConditions", joinConditions)
+                .itemIf("postJoinFilter", postJoinFilter, postJoinFilter != null)
+                .item("select", String.join(",", getRowType().getFieldNames()))
+                .item("rowType", getRowType());
+    }
+
+    @Override
+    protected RelDataType deriveRowType() {
+        return this.rowType;
+    }
+
+    @Override
+    public ExecNode<?> translateToExecNode() {
+        final RexNode multiJoinCondition = createMultiJoinCondition();
+        final List<List<int[]>> inputUpsertKeys = getUpsertKeysForInputs();
+        final List<FlinkJoinType> execJoinTypes = getExecJoinTypes();
+
+        return new StreamExecMultiJoin(
+                unwrapTableConfig(this),
+                execJoinTypes,
+                joinConditions,
+                multiJoinCondition,
+                joinAttributeMap,
+                inputUpsertKeys,
+                Collections.emptyMap(), // TODO Enable hint-based state ttl. See ticket
+                // TODO https://issues.apache.org/jira/browse/FLINK-37936
```

Comment:
not sure people who will review FLINK-37936 will be able to detect these TODOs...
should they be explicitly mentioned in JIRA?